src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -394,6 +395,48 @@ export class ManageExtensionAction extends Action {
 	}
 }
 
+export class AddToWorkspaceRecommendationsAction extends Action implements IExtensionAction {
+
+	static ID = 'extensions.addToWorkspaceRecommendationsAction';
+	static LABEL = localize('addToWorkspaceRecommendationsAction', "Workspace");
+
+	private disposables: IDisposable[] = [];
+
+	private _extension: IExtension;
+	get extension(): IExtension { return this._extension; }
+	set extension(extension: IExtension) { this._extension = extension; this.update(); }
+
+	constructor(label: string,
+		@IWorkspaceContextService private workspaceContextService: IWorkspaceContextService,
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService,
+		@IExtensionEnablementService private extensionEnablementService: IExtensionEnablementService,
+		@IInstantiationService private instantiationService: IInstantiationService
+	) {
+		super(AddToWorkspaceRecommendationsAction.ID, label);
+
+		this.disposables.push(this.extensionsWorkbenchService.onChange(() => this.update()));
+		this.update();
+	}
+
+	private update(): void {
+		this.enabled = !!this.extension;
```

Comment:
Enablement should also depends on if it is already recommended or not and also if there is a workspace or not